pkgs/by-name/co/copilot-language-server/package.nix

```diff
@@ -1,31 +1,55 @@
 {
   lib,
-  buildNpmPackage,
+  stdenv,
   fetchurl,
+  unzip,
+  nix-update-script,
 }:
 
-buildNpmPackage rec {
+let
+  arch =
+    {
+      aarch64-darwin = "arm64";
+      aarch64-linux = "arm64";
+      x86_64-darwin = "x64";
+      x86_64-linux = "x64";
+    }
+    ."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  os =
+    {
+      aarch64-darwin = "darwin";
+      aarch64-linux = "linux";
+      x86_64-darwin = "darwin";
+      x86_64-linux = "linux";
+    }
+    ."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+in
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "copilot-language-server";
   version = "1.275.0";
 
   src = fetchurl {
-    url = "https://registry.npmjs.org/@github/copilot-language-server/-/copilot-language-server-${version}.tgz";
-    hash = "sha256-OVqtwz9T5vSYAZc8nof0jXn7H40i1r7SAS6jK4xeSlo=";
+    url = "https://github.com/github/copilot-language-server-release/releases/download/${finalAttrs.version}/copilot-language-server-native-${finalAttrs.version}.zip";
+    hash = "sha256-ItvYpHRUuuxtUp2j4m28DSqoMHQlNvYqtE/J/Lfsiyw=";
   };
 
-  npmDepsHash = "sha256-PLX/mN7xu8gMh2BkkyTncP3+rJ3nBmX+pHxl0ONXbe4=";
+  nativeBuildInputs = [ unzip ];
 
-  postPatch = ''
-    ln -s ${./package-lock.json} package-lock.json
-  '';
+  sourceRoot = ".";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out"/bin
+    cp -a "${os}-${arch}"/copilot-language-server "$out"/bin/
```

Comment:
opinionated: use `install` and you don't need to `mkdir`. Basically, replace two commands with one single.

Reply:
It seems good! Done.